python/SoapySDR.i

```diff
@@ -33,21 +33,15 @@ See https://pothosware.github.io/SoapySDR/doxygen/latest/index.html for details.
 ////////////////////////////////////////////////////////////////////////
 // http://www.swig.org/Doc2.0/Library.html#Library_stl_exceptions
 ////////////////////////////////////////////////////////////////////////
-%include <exception.i>
+%include "swig_exceptions.i"
 
 // We only expect to throw DirectorExceptions from within
 // SoapySDR_pythonLogHandlerBase calls.  Catching them permits us to
 // propagate exceptions thrown in the Python log handler callback back to
 // Python.
-%exception
-{
-    try{$action}
-    catch (const Swig::DirectorException &e) { SWIG_fail; }
-    catch (const std::exception &ex)
-    {SWIG_exception(SWIG_RuntimeError, ex.what());}
-    catch (...)
-    {SWIG_exception(SWIG_RuntimeError, "unknown");}
-}
+SWIG_TRY
+catch (const Swig::DirectorException &e) { SWIG_fail; }
+SWIG_CATCH
```

Comment:
so the boost stuff doesnt really apply, the build isnt setting SWIG_WRAP_BOOST_EXCEPTIONS, so the real change here is that 

`catch (const std::exception &ex) {SWIG_exception(SWIG_RuntimeError, ex.what());}`

gets replaced by `SWIG_CATCH_STDEXCEPT`

Reply:
Functionally, I suppose that’s what it comes down to. I’m not sure how much of the “custom stuff looks nicer” we’ll actually need from the file I brought in, but I thought this was cleaner overall.